packages/extension-ui/src/Popup/Accounts/EditAccountMenu.tsx

```diff
@@ -3,41 +3,177 @@
+import React, { useCallback, useContext, useState } from 'react';
+import CopyToClipboard from 'react-copy-to-clipboard';
+import { RouteComponentProps, withRouter } from 'react-router';
 import styled from 'styled-components';
 
+import { AccountJson } from '@polkadot/extension-base/background/types';
+import { EditMenuCard, Identicon } from '@polkadot/extension-ui/components';
+import useMetadata from '@polkadot/extension-ui/hooks/useMetadata';
+import { IconTheme } from '@polkadot/react-identicon/types';
+
+import forgetIcon from '../../assets/forget.svg';
+import subAccountIcon from '../../assets/subAccount.svg';
+import { Switch } from '../../components';
+import { AccountContext, SettingsContext } from '../../components/contexts';
+import useToast from '../../hooks/useToast';
 import useTranslation from '../../hooks/useTranslation';
+import { showAccount } from '../../messaging';
 import Header from '../../partials/Header';
+import { ellipsisName } from '../../util/ellipsisName';
 
-interface Props extends ThemeProps {
+interface Props extends RouteComponentProps<{ address: string; isExternal: string }>, ThemeProps {
   className?: string;
 }
 
-function EditAccountMenu({ className }: Props): React.ReactElement<Props> {
+function EditAccountMenu({
+  className,
+  match: {
+    params: { address, isExternal }
+  }
+}: Props): React.ReactElement<Props> {
   const { t } = useTranslation();
+  const { accounts, hierarchy } = useContext(AccountContext);
+  const { show } = useToast();
 
-  return <>
-          <Header
-            showBackArrow
-            showHelp
-            text={t<string>('Edit Account')}
-        />
-          {/* TODO: */}
-        <div className={className}>PLACEHOLDER</div>
+  function findAccountInHierarchy(accounts: AccountJson[], _address: string) {
+    return hierarchy.find(({ address }): boolean => address === _address) || null;
+  }
+
+  const _onCopy = useCallback(() => show(t<string>('Public address copied to your clipboard'), 'success'), [show, t]);
+
+  const [account, setAccount] = useState(findAccountInHierarchy(accounts, address));
+
+  const [isHidden, setIsHidden] = useState(account?.isHidden);
+
+  console.log('account', account);
+
+  const chain = useMetadata(account?.genesisHash, true);
+
+  const settings = useContext(SettingsContext);
+
+  const prefix = chain ? chain.ss58Format : settings.prefix === -1 ? 42 : settings.prefix;
+
+  const theme = (account && account.type === 'ethereum' ? 'ethereum' : chain?.icon || 'polkadot') as IconTheme;
 
-  </>;
+  const _toggleVisibility = useCallback((): void => {
+    console.log('toggleVisibility from address', address, isHidden);
+    address &&
+      showAccount(address, isHidden || false)
+        .then((data) => {
+          if (account) {
+            setAccount({
+              ...account,
+              isHidden: !account.isHidden
+            });
+          }
+
+          setIsHidden(data);
+        })
+        .catch(console.error);
```

Comment:
As in one of the previous comments; code should be semantic. If you need to check for condition it should be via if statement not by logical operator `&&`.